### ISSUE TITLE
Don't require vendor autoloading for php-fpm

### ISSUE THREAD

**Issue by jasonmccallister** [brefphp/bref#557]
I have a simple function defined for a health check, its coming in from API Gateway but it is a very simple file that looks like this:

```php
<?php

try {
    (new PDO($dsn, $user, $password))
        ->setAttribute(PDO::ATTR_ERRMODE, PDO::ERRMODE_EXCEPTION);

    echo 'everything is good to go';
} catch (PDOException $e) {
    echo 'something went wrong';
}
```

In this case, I don't _need_ the vendor autoload but the default php-fpm layer is always attempting to load. 

Would a similar approach taken in #556 of adding an environment variable like `BREF_DISABLE_AUTOLOAD` make sense?

**Comment by mnapoli** [comment]
I see. The thing is that the FPM layer requires the autoloader because it uses classes from those dependencies (Bref classes):

https://github.com/brefphp/bref/blob/6893d7aa2f77d13892127a0b8f8c14f4d8af3863/runtime/layers/fpm/bootstrap#L4-L5

Without the `vendor` directory, the FPM layer cannot work.

We've discussed making the layers "self-contained" in the past, but it's a little bit of effort, it introduces a bit of complexity (in the code and release process), and I'm not really sure it's worth it to be honest.

What would be the gain from not deploying the vendor directory in your case?

**Comment by jasonmccallister** [comment]
That makes sense.

I am deploying a seperate app with its own dependencies (in a "monorepo" setup). This function is only checking the database connection and maybe other down the line. Right now I have to include the main or real applications vendor dependencies with that health check. It takes the 123kb file to over 10mb for that simple php code above.

**Comment by aknosis** [comment]
@jasonmccallister If you have a separate function definition for this health check can you get creative with your [packaging](https://serverless.com/framework/docs/providers/aws/guide/packaging/) to accomplish this?

```yml
functions:
  health-check:
    handler: healthcheck.php
    timeout: 28
    layers:
      - ${bref:layer.php-74-fpm}
    events:
      - http: 'ANY /'
      - http: 'ANY /{proxy+}'
package:
  exclude:
    - 'node_modules/**'
    - 'tests/**'
    - 'vendor/**'
  include:
    - 'vendor/bref**'
    - 'vendor/aws**'
```

I'm not 100% sure what's required to deploy, but in my testing my smallest artifacts with a hello world script were about 3MB.

While it is wasteful to deploy a "single file" in a 10MB package, I'm curious what impact does this have beyond that?

**Comment by mnapoli** [comment]
FYI #575 was merged, it removes the 18Mb AWS SDK that is installed with Bref by default. That should help a little bit, maybe even make the code editable inline in Lambda? (not sure, I haven't tested)

**Comment by jasonmccallister** [comment]
I did get creative with using a combination of layers after #556 so I still require the vendor path to @mnapoli's point, but I can edit the code inside of Lambda editor now.

**Comment by mnapoli** [comment]
OK, closing this for now as this isn't a goal at the moment. (we can of course revisit later)

**Comment by mnapoli** [comment]
Reopening for Bref 2.0 👀

**Comment by mnapoli** [comment]
Implemented in 2.0 (not released yet).

Bref 2.0 runtimes are here: https://github.com/brefphp/aws-lambda-layers

More details about the new runtimes, for those interested: https://github.com/brefphp/bref/issues/1299
